scripts/Test-RepoPackage.ps1

```diff
@@ -0,0 +1,633 @@
+  }
+}
+
+function RemoveDependentPackages() {
+  param(
+    [object]$packages
+  )
+
+  [array]$dependentPackages = $packages | ? { $_.DependentPackage -ne $null -and $_.DependentPackage -ne '' } | select -expand DependentPackage
+  if ($dependentPackages -ne $null -and $dependentPackages.Count -gt 0) {
+    $packages = $packages | ? { !$dependentPackages.Contains($_.Name) }
+  }
+  return $packages
+}
+
+function RunChocoPackProcess() {
+<#
+.SYNOPSIS
+  Function responsible for running choco pack when a nupkg file have not already been created.
+#>
+  param([string]$path)
+  if ($path -ne $null -and $path -ne '') { pushd $path }
+  if (!(Test-Path "*.nupkg")) {
+    . choco pack | WriteChocoOutput
+    if ($LastExitCode -ne 0) { popd; throw "Choco pack failed with code: $LastExitCode"; return }
+  }
+  if ($path -ne $null -and $path -ne '') { popd}
+}
+
+function SetAppveyorExitCode() {
+<#
+.SYNOPSIS
+  Sets the exit code of the script when running on appveyor, so
+  the build would fail when necessary.
+#>
+  param(
+    [int]$ExitCode
+  )
+  WriteOutput "Exit code was $ExitCode" -type Warning
+
+  if (!(Test-Path env:\APPVEYOR)) {
+    return
+  }
+  if ($ExitCode -ne 0) {
+    $host.SetShouldExit($ExitCode)
+  }
+}
+
+function UploadAppveyorArtifact() {
+<#
+.SYNOPSIS
+  Uploads the specified filePaths to appveyor when a appveyor build
+  is currently running.
+#>
+  param(
+    [string[]]$filePaths
+  )
+
+  if (Test-Path env:\APPVEYOR) {
+    $filePaths | ? { Test-Path $_ } | % { Push-AppveyorArtifact $_ -FileName (Split-Path -Leaf $_)}
+  }
+}
+
+function RunChocoProcess() {
+<#
+  Function responsible for running choco install/uninstall
+  and handling choco failures.
+#>
+  param(
+    [Parameter(Mandatory = $true)]
+    [string[]]$arguments,
+    [Parameter(Mandatory = $true)]
+    [int]$timeout,
+    [Parameter(Mandatory = $true)]
+    [bool]$takeScreenshot,
+    [Parameter(Mandatory = $true)]
+    [int]$timeoutBeforeScreenshot,
+    [Parameter(Mandatory = $true)]
+    [string]$screenShotDir
+  )
+
+  $res = $true
+  $args = @($arguments) + @(
+    '--yes'
+    "--execution-timeout=$timeout"
+  )
+  if ($arguments[0] -eq 'install') {
+    $args += @(
+      '--requirechecksum'
+    )
+  } else {
+    $args += @(
+      '--all-versions'
+      '--autouninstaller'
+      '--fail-on-autouninstaller'
+    )
+  }
+  $packFailed = $false
+  $errorFilePath = "$screenShotDir\$($arguments[0])Error_$($arguments[1]).jpg"
+  if (!(Test-Path "$screenShotDir")) { mkdir "$screenShotDir" -Force | Out-Null }
+
+  try {
+    RunChocoPackProcess '' | WriteChocoOutput
+    $failureOccurred = $false
+    $previousPercentage = -1;
+    $progressRegex = 'Progress\:.*\s+([\d]+)\%'
+    . choco $args | % {
+      $matches = $null
+      if ($failureOccurred) { WriteOutput $_ -type ChocoError }
+       # We are only showing progress per 10th value
+      elseif([regex]::IsMatch($_, $progressRegex)) {
```

Comment:
The chocolatey download progress.
I found during testing that it outputted way to much, so added changes to only output per 10th percentile.
ie `10% 20% 30% 40% 50%` etc. and only once per those percentages.